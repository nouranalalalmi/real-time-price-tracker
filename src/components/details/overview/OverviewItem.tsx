import { OverviewItemInterface } from './OverviewItems.interface';

export const OverviewItem = ({ label, children }: OverviewItemInterface) => (
  <div className="flex flex-col gap-y-1 text-xl text-gray-700">
    <h2 className="font-bold text-sm">{label}</h2>
    {children}
  </div>
);
