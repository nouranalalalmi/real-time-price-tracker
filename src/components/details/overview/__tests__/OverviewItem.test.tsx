import { render, screen } from '@testing-library/react';

import { OverviewItem } from '../OverviewItem';

describe('OverviewItem', () => {
  it('should render the label and children', () => {
    const { container } = render(
      <OverviewItem label="Test Label">
        <div>Test Child</div>
      </OverviewItem>
    );

    expect(screen.getByText('Test Label')).toBeInTheDocument();
    expect(screen.getByText('Test Child')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
