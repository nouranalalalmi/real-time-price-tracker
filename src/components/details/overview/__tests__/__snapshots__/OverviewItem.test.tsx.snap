// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewItem should render the label and children 1`] = `
<div>
  <div
    class="flex flex-col gap-y-1 break-all text-xl text-gray-700"
  >
    <h2
      class="text-sm font-bold"
    >
      Test Label
    </h2>
    <div>
      Test Child
    </div>
  </div>
</div>
`;
